torch/backends/cudnn/__init__.py

```diff
@@ -150,8 +167,8 @@ def set_flags(_enabled, _benchmark, _deterministic, _verbose):
                   verbose)
     verbose = _verbose
     torch._C._set_cudnn_enabled(_enabled)
-    torch._C._set_cudnn_benchmark(_benchmark)
-    torch._C._set_cudnn_deterministic(_deterministic)
+    torch._C._set_cudnn_benchmark(_benchmark if cuda is not None else False)
```

Comment:
If we don't add this guard for miopen, would it run into error? (or just like a no-op?)

Reply:
I think it would be a no-op as MIOpen does not have an equivalent for that functionality. I was trying to make sure the getter doesn't return `True` for those fields on ROCm lest the user get the impression that MIOpen is "broken" because the setter was clearly toggling the fields. If there's another way to convey clearly to the user that these fields would have no effect on ROCm/MIOpen, that'd be fine too. However, silently toggling the value of the field without it having any discernible effect on functionality seemed like it would be confusing from a user perspective.

Reply:
This is not the only place users can set cudnn benchmark, e.g. they can do `torch.backends.cudnn.benchmark = True`. If we want to catch and provide better warning/error message, a better place would be the underlying c++ `_set_cudnn_benchmark` (i.e. https://github.com/pytorch/pytorch/blob/fb159b5/torch/csrc/Module.cpp#L449-L455).

Reply:
I do need to clarify that I was partially wrong above:
1. The MIOpen path **does** currently support the deterministic mode.
2. The MIOpen path **doesn't** support `benchmark=False` mode, so the current implementation assumes `benchmark=True`.

Reply:
Added a warning message in the Module.cpp file.